vic/drivers/cesm/cpl_esmf/lnd_comp_esmf.F90

```diff
@@ -0,0 +1,178 @@
+module lnd_comp_esmf
+
+! !USES:
+
+  use ESMF
+  use esmfshr_mod
+!
+! !PUBLIC TYPES:
+  implicit none
+  save
+  private ! except
+
+!--------------------------------------------------------------------------
+! Public interfaces
+!--------------------------------------------------------------------------
+
+  public :: lnd_init_esmf
+  public :: lnd_run_esmf
+  public :: lnd_final_esmf
+  public :: lnd_register_esmf
+
+!--------------------------------------------------------------------------
+! Private data interfaces
+!--------------------------------------------------------------------------
+
+!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+CONTAINS
+!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+subroutine lnd_register_esmf(comp, rc)
+    type(ESMF_GridComp)  :: comp
+    integer, intent(out) :: rc
+
+    rc = ESMF_SUCCESS
+
+    print *, "In lnd register routine"
```

Comment:
@bartnijssen - I should have steered you away from the `cpl_esmf/*` sources.  We're not using these at the moment and I'm not sure we can completely remove them. All the coupling is done through `cpl_mct`.

@apcraig - Do we need to leaf the `cpl_esmf` directory in here for any testing purposes?  I have the boilerplate code in there right now but would prefer to remove it if it will never be used. If we need it for testing or compliance with CESM, I'll just leave it as is.